components/mention/mention.component.ts

```diff
@@ -322,13 +327,15 @@ export class NzMentionComponent implements OnInit, AfterViewInit, OnChanges {
   }
 
   private handleClick(): void {
-    this.resetDropdown();
+    this.resetDropdown(false);
```

Comment:
This take me a long time to investigate after the refactor, but finally i found the problem :) 

The issue was that after the signal refactor, both click and input events were calling resetDropdown() with emit: true, causing nzOnSearchChange to be emitted twice instead of once.

In the original service-based implementation, there was likely a timing issue or subscription management that prevented the click event from properly emitting the search change event,
 
Making the test expect only 1 call instead of the functionally correct 2 calls.

From my point of view, when user click on the input, nzOnSearch do not have to emit, that why i had false here.